example_test.go

```diff
@@ -62,6 +68,80 @@ func Example() {
 	}
 }
 
+type ExampleCRDWithConfigMapRef struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	ConfigMapRef      corev1.LocalObjectReference `json:"configMapRef"`
+}
+
+// DeepCopyObject implements client.Object.
+func (*ExampleCRDWithConfigMapRef) DeepCopyObject() runtime.Object {
+	panic("unimplemented")
+}
+
+type ExampleCRDWithConfigMapRefList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ExampleCRDWithConfigMapRef `json:"items"`
+}
+
+// DeepCopyObject implements client.ObjectList.
+func (*ExampleCRDWithConfigMapRefList) DeepCopyObject() runtime.Object {
+	panic("unimplemented")
+}
+
+// This example creates a simple application Controller that is configured for ExampleCRDWithConfigMapRef CRD.
+// Any change in the configMap referenced in this Custom Resource will cause the re-reconcile of the parent ExampleCRDWithConfigMapRef
+// due to the implementation of the .Watches method of "sigs.k8s.io/controller-runtime/pkg/builder".Builder.
+func Example_watches() {
```

Comment:
I don't think the fact that this has a watch is the special bit about this example. And I'd rather not have an example that doesn't implement `DeepCopy` but just panics. Maybe do something like a special configmap key that references something else or such?
```suggestion
func ExampleCustomHandler() {
```

Reply:
> I don't think the fact that this has a watch is the special bit about this example. 

That was my intention, to have an easy to grasp example for "Watches" 😢 

> Maybe do something like a special configmap key that references something else or such?

I didn't want to complicate this example too much, I thought about switching from LocalObjectReference to e.g [ConfigMapKeySelector](https://github.com/kubernetes/api/blob/v0.28.2/core/v1/types.go#L2137), but the code inside Watches would a too complicated for an example imho.

> And I'd rather not have an example that doesn't implement DeepCopy but just panics. 

Sure, I've changed it to json.Marshal + json.Unmarshal with panic on error, is that enough? I could use `controller-gen` for that, but that would result in new file in the root of the repo + some changes in makefile, which I wanted to omit